File: bioprocs/scripts/utils/helpers.params2CmdArgs.py
if 'params2CmdArgs' not in vars() or not callable(params2CmdArgs):
	# dash  = 'auto', '-', '--', ''
	# equal = 'auto', '=', ' '
	def params2CmdArgs(params, dash = 'auto', equal = 'auto', noq = None):
		try:  # py3
			from shlex import quote
		except ImportError:  # py2
			from pipes import quote
		ret = []
		for key, val in params.items():
			key = key.strip()
			if isinstance(val, bool) and not val: continue
			item = '--' if (dash == 'auto' and len(key) > 1) else '-' if dash == 'auto' else dash
			item += key
			if not isinstance(val, bool):
				item += '=' if (equal == 'auto' and len(key)>1) else ' ' if equal == 'auto' else equal
				if not noq or key not in list(noq):
					item += quote(str(val))
				else:
					item += str(val)
			ret.append(item)
		return ' '.join(ret)

File: bioprocs/scripts/utils/helpers.params2CmdArgs.r
if (!exists('params2CmdArgs')) {
	params2CmdArgs = function (params, dash = 'auto', equal = 'auto', noq = c()) {
		ret = c()
		for (key in names(params)) {
			val = params[[key]]
			key = trimws(key)
			if (is.logical(val) && !val) next
			item = if (dash == 'auto' && nchar(key) > 1) '--' else if (dash == 'auto') '-' else dash
			item = paste0(item, key)
			if (!is.logical(val)) {
				item = paste0(item, if (equal == 'auto' && nchar(key)>1) '=' else if (equal == 'auto') ' ' else equal)
				if (length(noq) == 0 || !(key %in% noq)) {
					item = paste0(item, shQuote(val))
				} else {
					item = paste0(item, val)
				}
			}
			ret = c(ret, item)
		}
		return(paste(ret, collapse=' '))
	}

}